File: frontend/src/types/job.ts
export interface Job {
  id: string
  type: string
  status: 'pending' | 'running' | 'completed' | 'failed'
  progress: number
  current_step: string | null
  result: string | null
  error: string | null
  created_at: number
  started_at: number | null
  completed_at: number | null
}

export interface ImportURLRequest {
  url: string
  library_id?: string | null
  user_metadata?: Record<string, any>
  imported_by?: string | null
  auto_mode?: boolean
}

export interface ImportURLResponse {
  success: boolean
  entry_uuid?: string
  inbox_id?: string
  inbox_type?: string
  job_id: string
}

File: backend/app/schemas/job.py
"""
Videorama v2.0.0 - Job Schemas
"""

from pydantic import BaseModel, Field, field_validator
from typing import Optional, Dict
import json


class JobCreate(BaseModel):
    """Schema for creating a job"""

    type: str = Field(
        ...,
        pattern="^(import|import_filesystem|move|enrich|reindex)$",
    )
    metadata: Optional[Dict] = None


class JobResponse(BaseModel):
    """Schema for job response"""

    id: str
    type: str
    status: str = Field(..., pattern="^(pending|running|completed|failed|cancelled)$")
    progress: float = Field(0.0, ge=0.0, le=1.0)
    current_step: Optional[str] = None
    result: Optional[Dict] = None
    error: Optional[str] = None
    created_at: float
    started_at: Optional[float] = None
    updated_at: Optional[float] = None
    completed_at: Optional[float] = None

    @field_validator('result', mode='before')
    @classmethod
    def parse_result(cls, v):
        """Parse result from JSON string if needed"""
        if v is None:
            return None
        if isinstance(v, str):
            try:
                return json.loads(v)
            except json.JSONDecodeError:
                return None
        return v

    class Config:
        from_attributes = True
